onnx/common/assertions.cc

```diff
@@ -21,9 +21,12 @@ std::string barf(const char* fmt, ...) {
   va_list args;
 
   va_start(args, fmt);
-  // Although vsnprintf might have vulnerability issue while using format string with overflowed length,
-  // it should be safe here to use fixed length for buffer "msg". No further checking is needed.
-  vsnprintf(msg, 2048, fmt, args);
+
+  // use fixed length for buffer "msg" to avoid buffer overflow
+  vsnprintf(msg, sizeof(msg) - 1, fmt, args);
```

Comment:
Is this relevant? 

> warning: do not implicitly decay an array into a pointer; consider using gsl::array_view or an explicit cast instead [cppcoreguidelines-pro-bounds-array-to-pointer-decay]

Reply:
will os.path.basename(path) avoid 'path' being pointed to location outside of cwd? Here is my experiment:
>>> path="/external_data/../../secret"
>>> import os
>>> path2 = os.path.normpath(path)
>>> path2
'\\secret'
>>> path3 = os.path.basename(path)
>>> path3
'secret'

Reply:
fixed the error with std::array.